ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/HardFork/Combinator/Ledger.hs

```diff
@@ -113,6 +113,42 @@ deriving anyclass instance
      CanHardFork xs
   => NoThunks (Ticked (LedgerState (HardForkBlock xs)))
 
+-- | = HFC ticking
+--
+-- This is the place where era transitions happen, ie when we have @st ::
+-- 'LedgerState' ('HardForkBlock' xs)@ and, for a valid @slotNo@,
+--
+-- @
+-- tickedSt :: 'Ticked' ('LedgerState' ('HardForkBlock' xs))
+-- tickedSt = 'applyChainTick' cfg slotNo st
+-- @
+--
+-- we have the corresponding era indices
+--
+-- @
+-- ixSt, ixTickedSt :: 'EraIndex' xs
+-- ixSt =
+--     'eraIndexFromNS' . State.'State.tip' . 'hardForkLedgerStatePerEra' $ st
+-- ixTickedState =
+--     'eraIndexFromNS' . State.'State.tip' . 'tickedHardForkLedgerStatePerEra' $ tickedSt
+-- @
+--
+-- and we always have
+--
+-- >>> ixSt <= ixTickedSt
+-- True
+--
+-- with inequality exactly if we ticked across an era boundary.
+--
+-- == Ticking alone can't reveal era transitions
```

Comment:
The wording of this subheading seems off to me:

> Ticking alone can't reveal era transitions

It seems like there are two things worth explaining/referencing:

- The transition must have been predicted by `singleEraTransition` at least one safe zone before it happened.

- Ticking is currently unable to change the output of `singleEraTransition`.

Right now, the text of this subheading could be interpreted as either of those two things. Is that ambiguity intentional?

Also, the text seems to call the first one a "reflection" of the second one, which I hesitate to agree with.

Reply:
> Ticking is currently unable to change the output of `singleEraTransition`.

This is what I wanted to talk about here in this subsection, but I didn't phrase it like this as you can't apply `singleEraTransition` to the output of ticking.

I didn't intend to talk about safe zones here (but that could definitely happen in a different subsection, will add that :+1:). In my mind, these two topics are orthogonal:

 - This subsection is about *how* era transitions can become known (currently, only via block application, not via ticking).
 - Safe zones are about *when* an era transition can become known.

> Also, the text seems to call the first one a "reflection" of the second one, which I hesitate to agree with.

Yeah, I didn't want to imply that, hopefully, separating safe zones to its own section will make that more clear.